### PR TITLE
nc-news-7

### PR DESCRIPTION
**Pull request by jaymckerracher**

### CODE REVIEW

File: __tests__/integration.test.js [hunk 1]
```diff
@@ -170,7 +170,64 @@ describe('/api/articles/:article_id/comments', () => {
...
+        .post('/api/articles/1/comments')
+        .send({})
+        .expect(400)
+        .then(({body}) => {
+            const {msg} = body;
+            expect(msg).toBe('Bad Request')
+        })
+    });
+    test('POST: 400 sends an appropriate status and error message if fields violate constraints', () => {
+        return request(app)
```
Comment (MKD01) [inline __tests__/integration.test.js]: This test description is not very clear. What does this actually mean?

File: __tests__/integration.test.js [hunk 2]
```diff
@@ -170,7 +170,64 @@ describe('/api/articles/:article_id/comments', () => {
...
+    });
+    test('POST: 404 sends an appropriate status and error message if the user does not exist', () => {
+        return request(app)
+        .post('/api/articles/1/comments')
+        .send({username: 'jaymckerracher', body: 'This is a new comment!'})
+        .expect(404)
+        .then(({body}) => {
+            const {msg} = body;
+            expect(msg).toBe('User Not Found')
+        })
```
Comment (MKD01) [inline __tests__/integration.test.js:229]: These are all really good tests, but there are a couple that should also be here.

What should happen when the user makes a request to an article id that doesnt exist?
What should happen when the user makes a request to an invalid article id?

File: controllers/comments-controller.js [hunk 3]
```diff
@@ -12,4 +12,23 @@ exports.getCommentsByArticle = (req, res, next) => {
...
         .catch(next)
+}
+
+exports.postCommentByArticle = (req, res, next) => {
+    const {article_id} = req.params;
+    const comment_body = req.body;
+    const {username, body} = comment_body;
+    
+    if (typeof username !== 'string' || username.length < 1 || typeof body !== 'string' || body.length < 1) {
+        res.status(400).send({msg: 'Bad Request'})
```
Comment (MKD01) [inline controllers/comments-controller.js]: This logic can and should be handled in the model file

File: endpoints.json [hunk 4]
```diff
@@ -62,6 +62,20 @@
       ]
     }
   },
+  "POST /api/articles/:article_id/comments": {
+    "description": "adds a comment that is related to one of the articles",
+    "queries": [],
+    "exampleResponse": {
```
Comment (MKD01) [inline endpoints.json:69]: Dont forget to include the exampleBody of what we expect the user to send us to actually create a new comment.

**Review by MKD01 — CHANGES_REQUESTED** [review]
Good work so far, but I would like to see the requested changes made to the tests before you merge this in, so please re-submit this pr once those changes have been made.

**Comment by saimakhanom** [comment]
All good to merge now! Just a suggestion to think about - we don't need to add a separate query to check the user exists as if they don't at the time of trying to post the comment, we'll get a PSQL error which can be handled. Saves us from having to do an additional query :)

**Pull request merged**